src/cfdp_core/pdu_directive.cpp

```diff
@@ -0,0 +1,58 @@
+#include <cfdp_core/pdu_directive.hpp>
+#include <cfdp_core/pdu_enums.hpp>
+#include <cfdp_core/pdu_exceptions.hpp>
+#include <cfdp_core/utils.hpp>
+
+#include <cstdint>
+#include <span>
+#include <vector>
+
+namespace header    = ::cfdp::pdu::header;
+namespace utils     = ::cfdp::utils;
+namespace exception = ::cfdp::pdu::exception;
+
+using ::cfdp::pdu::header::LargeFileFlag;
+
+cfdp::pdu::directive::KeepAlivePdu::KeepAlivePdu(uint64_t progress)
+    : progress(progress), largeFileFlag(LargeFileFlag::LargeFile){};
+
+cfdp::pdu::directive::KeepAlivePdu::KeepAlivePdu(uint32_t progress)
+    : progress(progress), largeFileFlag(LargeFileFlag::SmallFile){};
+
+cfdp::pdu::directive::KeepAlivePdu::KeepAlivePdu(std::span<uint8_t const> memory)
+{
+    const auto memory_size = memory.size();
+
+    if (memory_size < const_small_file_size_bytes)
+    {
+        throw exception::DecodeFromBytesException("Passed memory does not contain enough bytes");
+    }
+
+    if (memory[0] != (uint8_t)Directive::KeepAlive)
+    {
+        throw exception::DecodeFromBytesException("File Directive code is not Keep Alive Pdu");
+    }
+
+    largeFileFlag = (memory_size > const_small_file_size_bytes) ? LargeFileFlag::LargeFile
+                                                                : LargeFileFlag::SmallFile;
+    progress      = utils::bytesToInt<uint64_t>(
+        memory, 1,
+        (largeFileFlag == LargeFileFlag::LargeFile) ? sizeof(uint64_t) : sizeof(uint32_t));
```

Comment:
We can make a helper out of this size calculation, or even keep it as a member (you can think about it), because you use it in both encode and decode

Reply:
good idea, added private inline method